From: Till Kamppeter <till.kamppeter@gmail.com>
Date: Fri, 22 Jul 2016 09:33:04 +0200
Subject: Workaround patch for missing Python3 transition of the old
 (pre-USB-storage) photo memory card support (pcardext) as this part builds
 in Python3 environments but with pointer-related warnings which are fatal
 errors for Ubuntu's build servers. The patch silences the warnings but the
 memory card support is dropped in Python3 environments. This patch is
 supplied by the HPLIP upstream developers and will be replaced by a more
 proper solution in the next upstream release of HPLIP (see LP: #1275353)

---
 pcard/pcardext/pcardext.c | 59 +++++++++++++++++++++++++++++++++++++----------
 pcard/photocard.py        |  2 +-
 unload.py                 |  5 ++++
 3 files changed, 53 insertions(+), 13 deletions(-)

diff --git a/pcard/pcardext/pcardext.c b/pcard/pcardext/pcardext.c
index c1a8273..37d979b 100644
--- a/pcard/pcardext/pcardext.c
+++ b/pcard/pcardext/pcardext.c
@@ -20,7 +20,7 @@ pcardext - Python extension for HP photocard services
 Requires:
 Python 2.2+
 
-Author: Don Welch
+Author: Don Welch 
 
 \*****************************************************************************/
 
@@ -38,9 +38,37 @@ typedef int Py_ssize_t;
 
 int verbose=0;
 
+#if PY_MAJOR_VERSION >= 3
+  #define MOD_ERROR_VAL NULL
+  #define MOD_SUCCESS_VAL(val) val
+  #define MOD_INIT(name) PyMODINIT_FUNC PyInit_##name(void)
+  #define PyInt_AS_LONG PyLong_AS_LONG
+  #define MOD_DEF(ob, name, doc, methods) \
+          static struct PyModuleDef moduledef = { \
+            PyModuleDef_HEAD_INIT, name, doc, -1, methods, }; \
+          ob = PyModule_Create(&moduledef);
+
+
+  #define PY_String_Bytes  PyBytes_FromStringAndSize
+  #define PY_AsString_Bytes  PyBytes_AsStringAndSize
+
+#else
+  #define MOD_ERROR_VAL
+  #define MOD_SUCCESS_VAL(val)
+  #define MOD_INIT(name) void init##name(void)
+  #define MOD_DEF(ob, name, doc, methods)         \
+        ob = Py_InitModule3(name, methods, doc);
+
+  #define PY_String_Bytes PyString_FromStringAndSize
+  #define PY_AsString_Bytes PyString_AsStringAndSize
+  
+#endif
+
 PyObject * readsectorFunc = NULL;
 PyObject * writesectorFunc = NULL;
 
+
+
 int ReadSector(int sector, int nsector, void *buf, int size)
 {
     PyObject * result;
@@ -56,9 +84,13 @@ int ReadSector(int sector, int nsector, void *buf, int size)
         if( result )
         {
             Py_ssize_t len = 0;
-            PyString_AsStringAndSize( result, &result_str, &len );
+
+            //PyString_AsStringAndSize( result, &result_str, &len );    
+            //PyBytes_AsStringAndSize( result, &result_str, &len ); 
+            PY_AsString_Bytes( result, &result_str, &len );
             
-            if( len < nsector*FAT_HARDSECT )
+
+	    if( len < nsector*FAT_HARDSECT )
             {
                 goto abort;
             }
@@ -205,7 +237,9 @@ PyObject * pcardext_read( PyObject * self, PyObject * args )
     
     if( FatReadFileExt( name, offset, len, buffer ) == len )
     {
-        return PyString_FromStringAndSize( (char *)buffer, len );
+        // return PyString_FromStringAndSize( (char *)buffer, len );
+        return PY_String_Bytes( (char *)buffer, len );
+        // return PyBytes_FromStringAndSize( (char *)buffer, len );
     }
     else
     {
@@ -233,14 +267,15 @@ static PyMethodDef pcardext_methods[] =
 
 static char pcardext_documentation[] = "Python extension for HP photocard services";
 
-void initpcardext( void )
-{
-    PyObject * mod = Py_InitModule4( "pcardext", pcardext_methods, 
-                                     pcardext_documentation, (PyObject*)NULL, 
-                                     PYTHON_API_VERSION );
-                     
-    if (mod == NULL)
-      return;
+MOD_INIT(pcardext)  {
+
+  PyObject* mod ;
+  MOD_DEF(mod, "pcardext", pcardext_documentation, pcardext_methods);
+  if (mod == NULL)
+    return MOD_ERROR_VAL;
+
+  return MOD_SUCCESS_VAL(mod);
+
 }
 
 
diff --git a/pcard/photocard.py b/pcard/photocard.py
index 913dc4e..2c73158 100644
--- a/pcard/photocard.py
+++ b/pcard/photocard.py
@@ -30,7 +30,7 @@ from base.codes import *
 from base import device, utils, exif
 
 try:
-    from . import pcardext
+    import pcardext
 except ImportError:
     if not os.getenv("HPLIP_BUILD"):
         log.error("PCARDEXT could not be loaded. Please check HPLIP installation.")
diff --git a/unload.py b/unload.py
index 3fdd5a3..ce8b069 100755
--- a/unload.py
+++ b/unload.py
@@ -44,6 +44,11 @@ except ImportError:
 
 # Local
 from base.g import *
+from base.sixext import PY3
+if PY3:
+    log.error("This functionality is not spported in python3 environment.")
+    sys.exit(1)
+
 from base import device, utils, tui, module
 from prnt import cups
 
diff -Naur hplip-3.20.11/align.py hplip-3.20.11.new/align.py
--- hplip-3.20.11/align.py	2020-11-30 01:03:51.000000000 +0100
+++ hplip-3.20.11.new/align.py	2021-01-04 14:44:02.315072368 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2003-2015 HP Development Company, L.P.
diff -Naur hplip-3.20.11/base/codes.py hplip-3.20.11.new/base/codes.py
--- hplip-3.20.11/base/codes.py	2020-11-30 01:03:51.000000000 +0100
+++ hplip-3.20.11.new/base/codes.py	2021-01-04 14:44:03.279077254 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2003-2015 HP Development Company, L.P.
diff -Naur hplip-3.20.11/base/device.py hplip-3.20.11.new/base/device.py
--- hplip-3.20.11/base/device.py	2020-11-30 01:03:51.000000000 +0100
+++ hplip-3.20.11.new/base/device.py	2021-01-04 14:44:03.158076641 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2003-2015 HP Development Company, L.P.
diff -Naur hplip-3.20.11/base/dime.py hplip-3.20.11.new/base/dime.py
--- hplip-3.20.11/base/dime.py	2020-11-30 01:03:51.000000000 +0100
+++ hplip-3.20.11.new/base/dime.py	2021-01-04 14:44:03.222076965 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2003-2015 HP Development Company, L.P.
diff -Naur hplip-3.20.11/base/g.py hplip-3.20.11.new/base/g.py
--- hplip-3.20.11/base/g.py	2020-11-30 01:03:51.000000000 +0100
+++ hplip-3.20.11.new/base/g.py	2021-01-04 14:44:03.247077092 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2003-2015 HP Development Company, L.P.
diff -Naur hplip-3.20.11/base/imageprocessing.py hplip-3.20.11.new/base/imageprocessing.py
--- hplip-3.20.11/base/imageprocessing.py	2020-11-30 01:03:51.000000000 +0100
+++ hplip-3.20.11.new/base/imageprocessing.py	2021-01-04 14:44:03.165076676 +0100
@@ -1,5 +1,5 @@
 
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2003-2018 HP Development Company, L.P.
diff -Naur hplip-3.20.11/base/ldif.py hplip-3.20.11.new/base/ldif.py
--- hplip-3.20.11/base/ldif.py	2020-11-30 01:03:51.000000000 +0100
+++ hplip-3.20.11.new/base/ldif.py	2021-01-04 14:44:03.204076873 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 """
 ldif - generate and parse LDIF data (see RFC 2849)
 written by Michael Stroeder <michael@stroeder.com>
diff -Naur hplip-3.20.11/base/logger.py hplip-3.20.11.new/base/logger.py
--- hplip-3.20.11/base/logger.py	2020-11-30 01:03:51.000000000 +0100
+++ hplip-3.20.11.new/base/logger.py	2021-01-04 14:44:03.151076605 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2002-2015 HP Development Company, L.P.
diff -Naur hplip-3.20.11/base/magic.py hplip-3.20.11.new/base/magic.py
--- hplip-3.20.11/base/magic.py	2020-11-30 01:03:51.000000000 +0100
+++ hplip-3.20.11.new/base/magic.py	2021-01-04 14:44:03.198076843 +0100
@@ -259,8 +259,8 @@
     [0, 'string', '=', b'#! /usr/local/bin/python', 'application/x-python'],
     [0, 'string', '=', b'#! /usr/local/bin/python', 'application/x-python'],
     [0, 'string', '=', b'eval "exec /usr/local/bin/python', 'application/x-python'],
-    [0, 'string', '=', b'#!/usr/bin/env python', 'application/x-python'],
-    [0, 'string', '=', b'#! /usr/bin/env python', 'application/x-python'],
+    [0, 'string', '=', b'#!/usr/bin/env python3', 'application/x-python'],
+    [0, 'string', '=', b'#! /usr/bin/env python3', 'application/x-python'],
     [0, 'string', '=', b'#!/bin/rc', 'text/script'],
     [0, 'string', '=', b'#! /bin/rc', 'text/script'],
     [0, 'string', '=', b'#! /bin/rc', 'text/script'],
@@ -1247,7 +1247,7 @@
         return 'text/cpp'
 
     # its ASCII, now do text tests
-    if data.find(b'!/usr/bin/env python', 0, 256) > -1 or data.find(b'def ', 0, 8192) > -1:
+    if data.find(b'!/usr/bin/env python3', 0, 256) > -1 or data.find(b'def ', 0, 8192) > -1:
         return 'application/x-python'
 
     return "text/plain"
diff -Naur hplip-3.20.11/base/models.py hplip-3.20.11.new/base/models.py
--- hplip-3.20.11/base/models.py	2020-11-30 01:03:51.000000000 +0100
+++ hplip-3.20.11.new/base/models.py	2021-01-04 14:44:03.235077031 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2003-2015 HP Development Company, L.P.
diff -Naur hplip-3.20.11/base/password.py hplip-3.20.11.new/base/password.py
--- hplip-3.20.11/base/password.py	2020-11-30 01:03:51.000000000 +0100
+++ hplip-3.20.11.new/base/password.py	2021-01-04 14:44:03.184076772 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright @ 2015 HP Development Company, L.P.
diff -Naur hplip-3.20.11/base/pkit.py hplip-3.20.11.new/base/pkit.py
--- hplip-3.20.11/base/pkit.py	2020-11-30 01:03:51.000000000 +0100
+++ hplip-3.20.11.new/base/pkit.py	2021-01-04 14:44:03.266077188 +0100
@@ -337,7 +337,7 @@
     if utils.which("hp-plugin"):
         p_path="hp-plugin"
     else:
-        p_path="python ./plugin.py"
+        p_path="python3 ./plugin.py"
 
     cmd = "%s -u %s --reason %s" %(p_path, req, plugin_reason)   
     log.debug("%s" % cmd)
diff -Naur hplip-3.20.11/base/queues.py hplip-3.20.11.new/base/queues.py
--- hplip-3.20.11/base/queues.py	2020-11-30 01:03:51.000000000 +0100
+++ hplip-3.20.11.new/base/queues.py	2021-01-04 14:44:03.107076382 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2011-2015 HP Development Company, L.P.
diff -Naur hplip-3.20.11/base/smart_install.py hplip-3.20.11.new/base/smart_install.py
--- hplip-3.20.11/base/smart_install.py	2020-11-30 01:03:51.000000000 +0100
+++ hplip-3.20.11.new/base/smart_install.py	2021-01-04 14:44:03.145076574 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright @2015 HP Development Company, L.P.
diff -Naur hplip-3.20.11/base/status.py hplip-3.20.11.new/base/status.py
--- hplip-3.20.11/base/status.py	2020-11-30 01:03:51.000000000 +0100
+++ hplip-3.20.11.new/base/status.py	2021-01-04 14:44:03.139076544 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2003-2015 HP Development Company, L.P.
diff -Naur hplip-3.20.11/base/utils.py hplip-3.20.11.new/base/utils.py
--- hplip-3.20.11/base/utils.py	2020-11-30 01:03:51.000000000 +0100
+++ hplip-3.20.11.new/base/utils.py	2021-01-04 14:44:03.286077289 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2001-2018 HP Development Company, L.P.
@@ -585,7 +585,7 @@
         if len(path):
             self.cmd_pcard = 'hp-unload -d %DEVICE_URI%'
         else:
-            self.cmd_pcard = 'python %HOME%/unload.py -d %DEVICE_URI%'
+            self.cmd_pcard = 'python3 %HOME%/unload.py -d %DEVICE_URI%'
 
         # Copy
         path = which('hp-makecopies')
@@ -593,7 +593,7 @@
         if len(path):
             self.cmd_copy = 'hp-makecopies -d %DEVICE_URI%'
         else:
-            self.cmd_copy = 'python %HOME%/makecopies.py -d %DEVICE_URI%'
+            self.cmd_copy = 'python3 %HOME%/makecopies.py -d %DEVICE_URI%'
 
         # Fax
         path = which('hp-sendfax')
@@ -601,7 +601,7 @@
         if len(path):
             self.cmd_fax = 'hp-sendfax -d %FAX_URI%'
         else:
-            self.cmd_fax = 'python %HOME%/sendfax.py -d %FAX_URI%'
+            self.cmd_fax = 'python3 %HOME%/sendfax.py -d %FAX_URI%'
 
         # Fax Address Book
         path = which('hp-fab')
@@ -609,7 +609,7 @@
         if len(path):
             self.cmd_fab = 'hp-fab'
         else:
-            self.cmd_fab = 'python %HOME%/fab.py'
+            self.cmd_fab = 'python3 %HOME%/fab.py'
 
     def load(self):
         self.loadDefaults()
diff -Naur hplip-3.20.11/base/validation.py hplip-3.20.11.new/base/validation.py
--- hplip-3.20.11/base/validation.py	2020-11-30 01:03:51.000000000 +0100
+++ hplip-3.20.11.new/base/validation.py	2021-01-04 14:44:03.210076904 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2015 HP Development Company, L.P.
diff -Naur hplip-3.20.11/check-plugin.py hplip-3.20.11.new/check-plugin.py
--- hplip-3.20.11/check-plugin.py	2020-11-30 01:03:51.000000000 +0100
+++ hplip-3.20.11.new/check-plugin.py	2021-01-04 14:44:02.285072216 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2011-2015 HP Development Company, L.P.
diff -Naur hplip-3.20.11/check.py hplip-3.20.11.new/check.py
--- hplip-3.20.11/check.py	2020-11-30 01:03:51.000000000 +0100
+++ hplip-3.20.11.new/check.py	2021-01-04 14:44:02.419072896 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2003-2015 HP Development Company, L.P.
@@ -23,7 +23,7 @@
 __version__ = '15.1'
 __title__ = 'Dependency/Version Check Utility'
 __mod__ = 'hp-check'
-__doc__ = """Checks dependency versions,permissions of HPLIP. (Run as 'python ./check.py' from the HPLIP tarball before installation.)"""
+__doc__ = """Checks dependency versions,permissions of HPLIP. (Run as 'python3 ./check.py' from the HPLIP tarball before installation.)"""
 
 
 # Std Lib
diff -Naur hplip-3.20.11/clean.py hplip-3.20.11.new/clean.py
--- hplip-3.20.11/clean.py	2020-11-30 01:03:51.000000000 +0100
+++ hplip-3.20.11.new/clean.py	2021-01-04 14:44:02.303072307 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2003-2015 HP Development Company, L.P.
diff -Naur hplip-3.20.11/colorcal.py hplip-3.20.11.new/colorcal.py
--- hplip-3.20.11/colorcal.py	2020-11-30 01:03:51.000000000 +0100
+++ hplip-3.20.11.new/colorcal.py	2021-01-04 14:44:02.272072150 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2003-2015 HP Development Company, L.P.
diff -Naur hplip-3.20.11/config_usb_printer.py hplip-3.20.11.new/config_usb_printer.py
--- hplip-3.20.11/config_usb_printer.py	2020-11-30 01:03:51.000000000 +0100
+++ hplip-3.20.11.new/config_usb_printer.py	2021-01-04 14:44:02.406072830 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2011-2015 HP Development Company, L.P.
diff -Naur hplip-3.20.11/dat2drv.py hplip-3.20.11.new/dat2drv.py
--- hplip-3.20.11/dat2drv.py	2020-11-30 01:03:51.000000000 +0100
+++ hplip-3.20.11.new/dat2drv.py	2021-01-04 14:44:02.260072089 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2008-9 HP Development Company, L.P.
diff -Naur hplip-3.20.11/devicesettings.py hplip-3.20.11.new/devicesettings.py
--- hplip-3.20.11/devicesettings.py	2020-11-30 01:03:51.000000000 +0100
+++ hplip-3.20.11.new/devicesettings.py	2021-01-04 14:44:02.297072277 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2003-2015 HP Development Company, L.P.
diff -Naur hplip-3.20.11/diagnose_plugin.py hplip-3.20.11.new/diagnose_plugin.py
--- hplip-3.20.11/diagnose_plugin.py	2020-11-30 01:03:51.000000000 +0100
+++ hplip-3.20.11.new/diagnose_plugin.py	2021-01-04 14:44:02.229071932 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2015 HP Development Company, L.P.
diff -Naur hplip-3.20.11/diagnose_queues.py hplip-3.20.11.new/diagnose_queues.py
--- hplip-3.20.11/diagnose_queues.py	2020-11-30 01:03:51.000000000 +0100
+++ hplip-3.20.11.new/diagnose_queues.py	2021-01-04 14:44:02.376072678 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2011-2015 HP Development Company, L.P.
diff -Naur hplip-3.20.11/fab.py hplip-3.20.11.new/fab.py
--- hplip-3.20.11/fab.py	2020-11-30 01:03:51.000000000 +0100
+++ hplip-3.20.11.new/fab.py	2021-01-04 14:44:02.345072520 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2003-2015 HP Development Company, L.P.
diff -Naur hplip-3.20.11/fax/backend/hpfax.py hplip-3.20.11.new/fax/backend/hpfax.py
--- hplip-3.20.11/fax/backend/hpfax.py	2020-11-30 01:03:51.000000000 +0100
+++ hplip-3.20.11.new/fax/backend/hpfax.py	2021-01-04 14:44:03.013075906 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2003-2015 HP Development Company, L.P.
diff -Naur hplip-3.20.11/fax/soapfax.py hplip-3.20.11.new/fax/soapfax.py
--- hplip-3.20.11/fax/soapfax.py	2020-11-30 01:03:52.000000000 +0100
+++ hplip-3.20.11.new/fax/soapfax.py	2021-01-04 14:44:02.988075779 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2003-2015 HP Development Company, L.P.
diff -Naur hplip-3.20.11/faxsetup.py hplip-3.20.11.new/faxsetup.py
--- hplip-3.20.11/faxsetup.py	2020-11-30 01:03:52.000000000 +0100
+++ hplip-3.20.11.new/faxsetup.py	2021-01-04 14:44:02.364072617 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2003-2015 HP Development Company, L.P.
diff -Naur hplip-3.20.11/firmware.py hplip-3.20.11.new/firmware.py
--- hplip-3.20.11/firmware.py	2020-11-30 01:03:52.000000000 +0100
+++ hplip-3.20.11.new/firmware.py	2021-01-04 14:44:02.254072059 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2003-2015 HP Development Company, L.P.
diff -Naur hplip-3.20.11/hpdio.py hplip-3.20.11.new/hpdio.py
--- hplip-3.20.11/hpdio.py	2020-11-30 01:03:52.000000000 +0100
+++ hplip-3.20.11.new/hpdio.py	2021-01-04 14:44:02.248072028 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2003-2015 HP Development Company, L.P.
diff -Naur hplip-3.20.11/hpssd.py hplip-3.20.11.new/hpssd.py
--- hplip-3.20.11/hpssd.py	2020-11-30 01:03:52.000000000 +0100
+++ hplip-3.20.11.new/hpssd.py	2021-01-04 14:44:02.480073205 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2003-2015 HP Development Company, L.P.
diff -Naur hplip-3.20.11/info.py hplip-3.20.11.new/info.py
--- hplip-3.20.11/info.py	2020-11-30 01:03:52.000000000 +0100
+++ hplip-3.20.11.new/info.py	2021-01-04 14:44:02.449073048 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2003-2015 HP Development Company, L.P.
diff -Naur hplip-3.20.11/__init__.py hplip-3.20.11.new/__init__.py
--- hplip-3.20.11/__init__.py	2020-11-30 01:03:51.000000000 +0100
+++ hplip-3.20.11.new/__init__.py	2021-01-04 14:44:02.461073108 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2003-2015 HP Development Company, L.P.
diff -Naur hplip-3.20.11/installer/core_install.py hplip-3.20.11.new/installer/core_install.py
--- hplip-3.20.11/installer/core_install.py	2020-11-30 01:03:52.000000000 +0100
+++ hplip-3.20.11.new/installer/core_install.py	2021-01-04 14:44:03.033076007 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2003-2015 HP Development Company, L.P.
@@ -1961,7 +1961,7 @@
         cmds = []
         if self.get_distro_ver_data('fix_ppd_symlink', False, distro_ver):
             cmds.append(self.passwordObj.getAuthCmd() %
-                        'python ./installer/fix_symlink.py')
+                        'python3 ./installer/fix_symlink.py')
 
         return cmds
 
diff -Naur hplip-3.20.11/installer/dcheck.py hplip-3.20.11.new/installer/dcheck.py
--- hplip-3.20.11/installer/dcheck.py	2020-11-30 01:03:52.000000000 +0100
+++ hplip-3.20.11.new/installer/dcheck.py	2021-01-04 14:44:03.052076103 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2003-2015 HP Development Company, L.P.
diff -Naur hplip-3.20.11/installer/text_install.py hplip-3.20.11.new/installer/text_install.py
--- hplip-3.20.11/installer/text_install.py	2020-11-30 01:03:52.000000000 +0100
+++ hplip-3.20.11.new/installer/text_install.py	2021-01-04 14:44:03.040076042 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2003-2015 HP Development Company, L.P.
diff -Naur hplip-3.20.11/install.py hplip-3.20.11.new/install.py
--- hplip-3.20.11/install.py	2020-11-30 01:03:52.000000000 +0100
+++ hplip-3.20.11.new/install.py	2021-01-04 14:44:02.437072987 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2003-2015 HP Development Company, L.P.
diff -Naur hplip-3.20.11/levels.py hplip-3.20.11.new/levels.py
--- hplip-3.20.11/levels.py	2020-11-30 01:03:52.000000000 +0100
+++ hplip-3.20.11.new/levels.py	2021-01-04 14:44:02.351072551 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2003-2015 HP Development Company, L.P.
diff -Naur hplip-3.20.11/linefeedcal.py hplip-3.20.11.new/linefeedcal.py
--- hplip-3.20.11/linefeedcal.py	2020-11-30 01:03:52.000000000 +0100
+++ hplip-3.20.11.new/linefeedcal.py	2021-01-04 14:44:02.327072429 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2003-2015 HP Development Company, L.P.
diff -Naur hplip-3.20.11/makecopies.py hplip-3.20.11.new/makecopies.py
--- hplip-3.20.11/makecopies.py	2020-11-30 01:03:52.000000000 +0100
+++ hplip-3.20.11.new/makecopies.py	2021-01-04 14:44:02.456073083 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2003-2015 HP Development Company, L.P.
diff -Naur hplip-3.20.11/makeuri.py hplip-3.20.11.new/makeuri.py
--- hplip-3.20.11/makeuri.py	2020-11-30 01:03:52.000000000 +0100
+++ hplip-3.20.11.new/makeuri.py	2021-01-04 14:44:02.431072956 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2003-2015 HP Development Company, L.P.
diff -Naur hplip-3.20.11/pkservice.py hplip-3.20.11.new/pkservice.py
--- hplip-3.20.11/pkservice.py	2020-11-30 01:03:52.000000000 +0100
+++ hplip-3.20.11.new/pkservice.py	2021-01-04 14:44:02.412072860 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2003-2015 HP Development Company, L.P.
diff -Naur hplip-3.20.11/plugin.py hplip-3.20.11.new/plugin.py
--- hplip-3.20.11/plugin.py	2020-11-30 01:03:52.000000000 +0100
+++ hplip-3.20.11.new/plugin.py	2021-01-04 14:44:02.242071998 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2003-2015 HP Development Company, L.P.
diff -Naur hplip-3.20.11/pqdiag.py hplip-3.20.11.new/pqdiag.py
--- hplip-3.20.11/pqdiag.py	2020-11-30 01:03:52.000000000 +0100
+++ hplip-3.20.11.new/pqdiag.py	2021-01-04 14:44:02.339072490 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2003-2015 HP Development Company, L.P.
diff -Naur hplip-3.20.11/print.py hplip-3.20.11.new/print.py
--- hplip-3.20.11/print.py	2020-11-30 01:03:52.000000000 +0100
+++ hplip-3.20.11.new/print.py	2021-01-04 14:44:02.291072246 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2003-2015 HP Development Company, L.P.
diff -Naur hplip-3.20.11/printsettings.py hplip-3.20.11.new/printsettings.py
--- hplip-3.20.11/printsettings.py	2020-11-30 01:03:52.000000000 +0100
+++ hplip-3.20.11.new/printsettings.py	2021-01-04 14:44:02.388072738 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2003-2015 HP Development Company, L.P.
diff -Naur hplip-3.20.11/prnt/cups.py hplip-3.20.11.new/prnt/cups.py
--- hplip-3.20.11/prnt/cups.py	2020-11-30 01:03:52.000000000 +0100
+++ hplip-3.20.11.new/prnt/cups.py	2021-01-04 14:44:02.921075439 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2003-2015 HP Development Company, L.P.
diff -Naur hplip-3.20.11/probe.py hplip-3.20.11.new/probe.py
--- hplip-3.20.11/probe.py	2020-11-30 01:03:52.000000000 +0100
+++ hplip-3.20.11.new/probe.py	2021-01-04 14:44:02.370072647 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2003-2015 HP Development Company, L.P.
diff -Naur hplip-3.20.11/query.py hplip-3.20.11.new/query.py
--- hplip-3.20.11/query.py	2020-11-30 01:03:52.000000000 +0100
+++ hplip-3.20.11.new/query.py	2021-01-04 14:44:02.382072708 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2015 HP Development Company, L.P.
diff -Naur hplip-3.20.11/scan/sane.py hplip-3.20.11.new/scan/sane.py
--- hplip-3.20.11/scan/sane.py	2020-11-30 01:03:52.000000000 +0100
+++ hplip-3.20.11.new/scan/sane.py	2021-01-04 14:44:03.019075936 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2003-2015 HP Development Company, L.P.
diff -Naur hplip-3.20.11/scan.py hplip-3.20.11.new/scan.py
--- hplip-3.20.11/scan.py	2020-11-30 01:03:52.000000000 +0100
+++ hplip-3.20.11.new/scan.py	2021-01-04 14:44:02.223071902 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2003-2015 HP Development Company, L.P.
@@ -1916,7 +1916,7 @@
             elif d == 'print':
                 hp_print = utils.which("hp-print", True)
                 if not hp_print:
-                    hp_print = 'python ./print.py'
+                    hp_print = 'python3 ./print.py'
                  
                 if dest_printer is not None:
                    cmd = '%s -p %s %s &' % (hp_print, dest_printer, output)
diff -Naur hplip-3.20.11/sendfax.py hplip-3.20.11.new/sendfax.py
--- hplip-3.20.11/sendfax.py	2020-11-30 01:03:52.000000000 +0100
+++ hplip-3.20.11.new/sendfax.py	2021-01-04 14:44:02.279072186 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2003-2015 HP Development Company, L.P.
diff -Naur hplip-3.20.11/setup.py hplip-3.20.11.new/setup.py
--- hplip-3.20.11/setup.py	2020-11-30 01:03:52.000000000 +0100
+++ hplip-3.20.11.new/setup.py	2021-01-04 14:44:02.443073017 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2003-2015 HP Development Company, L.P.
@@ -919,7 +919,7 @@
                 if len(path) > 0:
                     cmd = 'hp-testpage -i %s' % param
                 else:
-                    cmd = 'python ./testpage.py -i %s' % param
+                    cmd = 'python3 ./testpage.py -i %s' % param
 
                 os_utils.execute(cmd)
 
diff -Naur hplip-3.20.11/systray.py hplip-3.20.11.new/systray.py
--- hplip-3.20.11/systray.py	2020-11-30 01:03:52.000000000 +0100
+++ hplip-3.20.11.new/systray.py	2021-01-04 14:44:02.473073169 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2003-2015 HP Development Company, L.P.
diff -Naur hplip-3.20.11/testpage.py hplip-3.20.11.new/testpage.py
--- hplip-3.20.11/testpage.py	2020-11-30 01:03:52.000000000 +0100
+++ hplip-3.20.11.new/testpage.py	2021-01-04 14:44:02.310072343 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2003-2015 HP Development Company, L.P.
diff -Naur hplip-3.20.11/timedate.py hplip-3.20.11.new/timedate.py
--- hplip-3.20.11/timedate.py	2020-11-30 01:03:52.000000000 +0100
+++ hplip-3.20.11.new/timedate.py	2021-01-04 14:44:02.334072464 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2003-2015 HP Development Company, L.P.
diff -Naur hplip-3.20.11/toolbox.py hplip-3.20.11.new/toolbox.py
--- hplip-3.20.11/toolbox.py	2020-11-30 01:03:52.000000000 +0100
+++ hplip-3.20.11.new/toolbox.py	2021-01-04 14:44:02.400072799 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2003-2015 HP Development Company, L.P.
diff -Naur hplip-3.20.11/ui/devmgr4.py hplip-3.20.11.new/ui/devmgr4.py
--- hplip-3.20.11/ui/devmgr4.py	2020-11-30 01:03:52.000000000 +0100
+++ hplip-3.20.11.new/ui/devmgr4.py	2021-01-04 14:44:03.329077507 +0100
@@ -2842,7 +2842,7 @@
         if utils.which('hp-setup'):
             cmd = 'hp-setup -u'
         else:
-            cmd = 'python ./setup.py --gui'
+            cmd = 'python3 ./setup.py --gui'
 
         log.debug(cmd)
         utils.run(cmd)
diff -Naur hplip-3.20.11/ui/nodevicesform.py hplip-3.20.11.new/ui/nodevicesform.py
--- hplip-3.20.11/ui/nodevicesform.py	2020-11-30 01:03:52.000000000 +0100
+++ hplip-3.20.11.new/ui/nodevicesform.py	2021-01-04 14:44:03.504078394 +0100
@@ -56,7 +56,7 @@
         if utils.which('hp-setup'):
             cmd = 'hp-setup -u'
         else:
-            cmd = 'python ./setup.py -u'
+            cmd = 'python3 ./setup.py -u'
 
         log.debug(cmd)
         utils.run(cmd)
diff -Naur hplip-3.20.11/ui/systemtray.py hplip-3.20.11.new/ui/systemtray.py
--- hplip-3.20.11/ui/systemtray.py	2020-11-30 01:03:52.000000000 +0100
+++ hplip-3.20.11.new/ui/systemtray.py	2021-01-04 14:44:03.486078303 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2003-2015 HP Development Company, L.P.
diff -Naur hplip-3.20.11/ui4/devmgr5.py hplip-3.20.11.new/ui4/devmgr5.py
--- hplip-3.20.11/ui4/devmgr5.py	2020-11-30 01:03:52.000000000 +0100
+++ hplip-3.20.11.new/ui4/devmgr5.py	2021-01-04 14:44:02.692074279 +0100
@@ -2098,7 +2098,7 @@
         if utils.which('hp-setup'):
             cmd = 'hp-setup --gui'
         else:
-            cmd = 'python ./setup.py --gui'
+            cmd = 'python3 ./setup.py --gui'
 
         log.debug(cmd)
         utils.run(cmd)
@@ -2110,7 +2110,7 @@
         if utils.which('hp-setup'):
             cmd = 'hp-setup --gui --remove'
         else:
-            cmd = 'python ./setup.py --gui --remove'
+            cmd = 'python3 ./setup.py --gui --remove'
 
         if self.cur_device_uri is not None:
             cmd += ' --device=%s' % self.cur_device_uri
@@ -2125,7 +2125,7 @@
         if utils.which('hp-diagnose_queues'):
             cmd= 'hp-diagnose_queues --gui'
         else:
-            cmd= 'python ./diagnose_queues.py --gui'
+            cmd= 'python3 ./diagnose_queues.py --gui'
         log.debug(cmd)
 #        ok, output = utils.run(cmd)
         os_utils.execute(cmd)
@@ -2135,7 +2135,7 @@
         if utils.which('hp-doctor'):
             cmd = 'hp-doctor -i -w'
         else:
-            cmd = 'python ./doctor.py -i -w'
+            cmd = 'python3 ./doctor.py -i -w'
 
         terminal_cmd = utils.get_terminal()
         if terminal_cmd:
diff -Naur hplip-3.20.11/ui4/nodevicesdialog.py hplip-3.20.11.new/ui4/nodevicesdialog.py
--- hplip-3.20.11/ui4/nodevicesdialog.py	2020-11-30 01:03:52.000000000 +0100
+++ hplip-3.20.11.new/ui4/nodevicesdialog.py	2021-01-04 14:44:02.598073803 +0100
@@ -53,7 +53,7 @@
         if utils.which('hp-setup'):
             cmd = 'hp-setup -u'
         else:
-            cmd = 'python ./setup.py -u'
+            cmd = 'python3 ./setup.py -u'
 
         log.debug(cmd)
         utils.run(cmd)
diff -Naur hplip-3.20.11/ui4/scandialog.py hplip-3.20.11.new/ui4/scandialog.py
--- hplip-3.20.11/ui4/scandialog.py	2020-11-30 01:03:52.000000000 +0100
+++ hplip-3.20.11.new/ui4/scandialog.py	2021-01-04 14:44:02.573073676 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 
 # Form implementation generated from reading ui file 'Hp-Scan.ui'
diff -Naur hplip-3.20.11/ui4/setupdialog.py hplip-3.20.11.new/ui4/setupdialog.py
--- hplip-3.20.11/ui4/setupdialog.py	2020-11-30 01:03:52.000000000 +0100
+++ hplip-3.20.11.new/ui4/setupdialog.py	2021-01-04 14:44:02.731074477 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2001-2015 HP Development Company, L.P.
diff -Naur hplip-3.20.11/ui4/systemtray.py hplip-3.20.11.new/ui4/systemtray.py
--- hplip-3.20.11/ui4/systemtray.py	2020-11-30 01:03:52.000000000 +0100
+++ hplip-3.20.11.new/ui4/systemtray.py	2021-01-04 14:44:02.548073549 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2003-2015 HP Development Company, L.P.
diff -Naur hplip-3.20.11/ui5/devmgr5.py hplip-3.20.11.new/ui5/devmgr5.py
--- hplip-3.20.11/ui5/devmgr5.py	2020-11-30 01:03:52.000000000 +0100
+++ hplip-3.20.11.new/ui5/devmgr5.py	2021-01-04 14:44:03.931080558 +0100
@@ -2143,7 +2143,7 @@
         if utils.which('hp-setup'):
             cmd = 'hp-setup --gui'
         else:
-            cmd = 'python ./setup.py --gui'
+            cmd = 'python3 ./setup.py --gui'
 
         log.debug(cmd)
         utils.run(cmd)
@@ -2155,7 +2155,7 @@
         if utils.which('hp-setup'):
             cmd = 'hp-setup --gui --remove'
         else:
-            cmd = 'python ./setup.py --gui --remove'
+            cmd = 'python3 ./setup.py --gui --remove'
 
         if self.cur_device_uri is not None:
             cmd += ' --device=%s' % self.cur_device_uri
@@ -2170,7 +2170,7 @@
         if utils.which('hp-diagnose_queues'):
             cmd= 'hp-diagnose_queues --gui'
         else:
-            cmd= 'python ./diagnose_queues.py --gui'
+            cmd= 'python3 ./diagnose_queues.py --gui'
         log.debug(cmd)
 #        ok, output = utils.run(cmd)
         os_utils.execute(cmd)
@@ -2180,7 +2180,7 @@
         if utils.which('hp-doctor'):
             cmd = 'hp-doctor -i -w'
         else:
-            cmd = 'python ./doctor.py -i -w'
+            cmd = 'python3 ./doctor.py -i -w'
 
         terminal_cmd = utils.get_terminal()
         if terminal_cmd:
diff -Naur hplip-3.20.11/ui5/nodevicesdialog.py hplip-3.20.11.new/ui5/nodevicesdialog.py
--- hplip-3.20.11/ui5/nodevicesdialog.py	2020-11-30 01:03:52.000000000 +0100
+++ hplip-3.20.11.new/ui5/nodevicesdialog.py	2021-01-04 14:44:03.839080092 +0100
@@ -53,7 +53,7 @@
         if utils.which('hp-setup'):
             cmd = 'hp-setup -u'
         else:
-            cmd = 'python ./setup.py -u'
+            cmd = 'python3 ./setup.py -u'
 
         log.debug(cmd)
         utils.run(cmd)
diff -Naur hplip-3.20.11/ui5/scandialog.py hplip-3.20.11.new/ui5/scandialog.py
--- hplip-3.20.11/ui5/scandialog.py	2020-11-30 01:03:52.000000000 +0100
+++ hplip-3.20.11.new/ui5/scandialog.py	2021-01-04 14:44:03.815079970 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 
 # Form implementation generated from reading ui file 'Hp-Scan.ui'
diff -Naur hplip-3.20.11/ui5/setupdialog.py hplip-3.20.11.new/ui5/setupdialog.py
--- hplip-3.20.11/ui5/setupdialog.py	2020-11-30 01:03:52.000000000 +0100
+++ hplip-3.20.11.new/ui5/setupdialog.py	2021-01-04 14:44:03.974080776 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2001-2015 HP Development Company, L.P.
diff -Naur hplip-3.20.11/ui5/systemtray.py hplip-3.20.11.new/ui5/systemtray.py
--- hplip-3.20.11/ui5/systemtray.py	2020-11-30 01:03:52.000000000 +0100
+++ hplip-3.20.11.new/ui5/systemtray.py	2021-01-04 14:44:04.211081977 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2003-2015 HP Development Company, L.P.
diff -Naur hplip-3.20.11/uiscan.py hplip-3.20.11.new/uiscan.py
--- hplip-3.20.11/uiscan.py	2020-11-30 01:03:52.000000000 +0100
+++ hplip-3.20.11.new/uiscan.py	2021-01-04 14:44:02.425072926 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 __version__ = '2.2'
 __mod__ = 'hp-uiscan'
 __title__ = 'Scan Utility'
diff -Naur hplip-3.20.11/unload.py hplip-3.20.11.new/unload.py
--- hplip-3.20.11/unload.py	2020-11-30 01:03:52.000000000 +0100
+++ hplip-3.20.11.new/unload.py	2021-01-04 14:44:02.394072769 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2003-2015 HP Development Company, L.P.
diff -Naur hplip-3.20.11/wificonfig.py hplip-3.20.11.new/wificonfig.py
--- hplip-3.20.11/wificonfig.py	2020-11-30 01:03:52.000000000 +0100
+++ hplip-3.20.11.new/wificonfig.py	2021-01-04 14:44:02.321072398 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # -*- coding: utf-8 -*-
 #
 # (c) Copyright 2003-2015 HP Development Company, L.P.
